lib/mayaUsd/ufe/UsdSceneItem.h

```diff
@@ -42,10 +46,21 @@ class MAYAUSD_CORE_PUBLIC UsdSceneItem : public Ufe::SceneItem
     UsdSceneItem& operator=(UsdSceneItem&&) = delete;
 
     //! Create a UsdSceneItem from a UFE path and a USD prim.
-    static UsdSceneItem::Ptr create(const Ufe::Path& path, const UsdPrim& prim);
+    static UsdSceneItem::Ptr create(
+        const Ufe::Path& path,
+        const UsdPrim& prim,
+        int instanceIndex = UsdImagingDelegate::ALL_INSTANCES);
 
     const UsdPrim& prim() const { return fPrim; }
 
+    void setInstanceIndex(int instanceIndex) {
+        _instanceIndex = instanceIndex;
+    }
+
+    int instanceIndex() const {
+        return _instanceIndex;
+    }
```

Comment:
I'm not convinced at all that this is the best way to handle storing and retrieving the `instanceIndex`, but this at least enables this prototype of selection highlighting.

Reply:
Is an instance index something that can be associated with any prim?  Only certain kinds of prims?  Perhaps this should be a derived class?  Or maybe we can have some kind of TfToken, VtValue pair map associated with a UsdSceneItem and an instance index would be one such pair?